src/ott/solvers/nn/neuraldual.py

```diff
@@ -327,3 +600,124 @@ def _penalize_weights_icnn(params: Dict[str, jnp.ndarray]) -> float:
       if k.startswith("w_z"):
         penalty += jnp.linalg.norm(jax.nn.relu(-param["kernel"]))
     return penalty
+
+
+def plot_ot_map(
+    learned_potentials: potentials.DualPotentials,
+    source: jnp.ndarray,
+    target: jnp.ndarray,
+    inverse: bool = False
+) -> Tuple[matplotlib.figure.Figure, matplotlib.axes.Axes]:
+  """Plot data and learned optimal transport map.
+
+  Args:
+    learned_potentials: the potentials object for the map
+    source: samples from the source measure
+    target: samples from the target measure
+    inverse: if the inverse map from the potentials
+    should be used
+
+  Returns: matplotlib figure and axis with the plots
+  """
+
+  def draw_arrows(a, b):
+    plt.arrow(
+        a[0], a[1], b[0] - a[0], b[1] - a[1], color=[0.5, 0.5, 1], alpha=0.3
+    )
+
+  grad_state_s = learned_potentials.transport(source, forward=not inverse)
+
+  fig = plt.figure(facecolor="white")
+  ax = fig.add_subplot(111)
+
+  if not inverse:
+    ax.scatter(
+        target[:, 0],
+        target[:, 1],
+        color="#A7BED3",
+        alpha=0.5,
+        label=r"$target$",
+    )
+    ax.scatter(
+        source[:, 0],
+        source[:, 1],
+        color="#1A254B",
+        alpha=0.5,
+        label=r"$source$",
+    )
+    ax.scatter(
+        grad_state_s[:, 0],
+        grad_state_s[:, 1],
+        color="#F2545B",
+        alpha=0.5,
+        label=r"$\nabla f(source)$",
+    )
+  else:
+    ax.scatter(
+        target[:, 0],
+        target[:, 1],
+        color="#A7BED3",
+        alpha=0.5,
+        label=r"$source$",
+    )
+    ax.scatter(
+        source[:, 0],
+        source[:, 1],
+        color="#1A254B",
+        alpha=0.5,
+        label=r"$target$",
+    )
+    ax.scatter(
+        grad_state_s[:, 0],
+        grad_state_s[:, 1],
+        color="#F2545B",
+        alpha=0.5,
+        label=r"$\nabla g(target)$",
+    )
+
+  fig.legend(ncol=3, loc="upper center")
```

Comment:
`ax.legend`

Reply:
`fig.legend` puts it above the axis:

![image](https://user-images.githubusercontent.com/707462/217826706-0eb8b2dc-71f0-4e82-8334-d6f531c13104.png)

but `ax.legend` puts it inside:

![image](https://user-images.githubusercontent.com/707462/217828658-74c64f5d-7238-45de-a69a-ac1a59b987df.png)

I'll keep `fig.legend` unless you meant that you prefer to have it inside